New tanks for newbs

Derplime opened the pull request:
**Description**
Updated registerUser API to give newly registered users an initial tank.

Resolves #243 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
1. Register a new user.
2. Verify on MongoDB Compass that an initial tank has been made owned by the user's id.

**Final Checklist:**
Go down the list and check them off.

Passes Flow:
![image](https://user-images.githubusercontent.com/24559442/77489414-6fa67980-6e0e-11ea-94eb-3e86615a510b.png)

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

Derplime committed:
Added logic to registerUser API to create a starter tank for them.

Derplime committed:
Weird spacing fix?

SecondThread commented on src/backend/controllers/userController.js:
```diff
@@ -75,6 +76,33 @@ exports.register = async (req: $Request, res: $Response) => {
+			}
+		});
+
+		const initTank = new Tank({
```
Did you check opening up Casus for this? I think you might need to have a container block with an empty block inside as the default casus block for things to work correctly

Derplime replied:
Oh do I? Alright. I'll double check. Mind putting what the blank container will look like in a comment below me?

blaiadam replied:
I think this is what he is referring to.

function getTestTank(id: number=1): Tank {
	const position=id===1?new Vec(20, -20):new Vec(50, 40);
	const toReturn: Tank = new Tank(
		position,
		new Chassis(id===1?'CHASSIS_5':'CHASSIS_2'), 
		new Treads(id===1?'TREAD_2':'TREAD_1'), 
		new Gun(id===1?'GUN_3':'GUN_4', false),
		new Gun(id===1?'GUN_9':'GUN_10', true),
		new Scanner(false, false, id===1?'MEDIUM':'SMALL'),
		new Jammer(id===1?'MEDIUM':'LARGE'),
		getEmptyCasusCode()
	);
	loadCasus(blocks => {toReturn.casusCode = blocks});
	return toReturn;
}

function getEmptyCasusCode(): CasusBlock {
	const childrenBlocks: Array<CasusBlock> = [];
	const container: ContainerBlock = new ContainerBlock(childrenBlocks);
	return container;
}

blaiadam replied:
Lol it wouldn't print it in code view. Check TankLoader.js in master.

SecondThread replied:
That might work? You would need to save the stringified version of that probably since this is after is is saved, but yeah, that's pretty much what it would look like. You can just use the default constructor (you don't need to pass in an empty array).

Derplime replied:
I know what to do.  I'll just use the code from SittingDuck. Just have to export the tanks and copy the casusCode structure from the JSON file

SecondThread reviewed (commented):
Just one thing to double check, but if that works, then the code looks good to me

Derplime committed:
Revised logic for adding initial tank and put base Casus code for blank area.

Derplime commented:
Updated code with better logic and added base Casus code with empty bounding box. For what it's worth, opening Casus Editor was fine with both the empty bounding box and the null values for casusCode, but stuck with the bounding box since it's at least "something" that's populating the field.

blaiadam commented on src/backend/controllers/userController.js:
```diff
@@ -66,12 +67,36 @@ exports.register = async (req: $Request, res: $Response) => {
+		let tank = new Tank();
+		tank.userId = user.id;
+		tank.tankName = 'Starter Tank';
+		tank.components = ['moddable', null, null, null, null, null, 'fastTreads', null, null, null];
```
Change the null to 'empty'.

Derplime committed:
empty instead of null

Derplime committed:
just one more empty

blaiadam reviewed (approved):
Thanks
